src/api/ledger/parse/ledger.js

```diff
@@ -0,0 +1,50 @@
+/* @flow */
+'use strict';
+const _ = require('lodash');
+const removeUndefined = require('./utils').removeUndefined;
+const parseTransaction = require('./transaction');
+
+function parseTransactions(transactions) {
+  if (!transactions || transactions.length === 0) {
+    return {};
+  }
+  if (typeof transactions[0] === 'string') {
+    return {transactionHashes: transactions};
+  }
+  return {
+    transactions: _.map(transactions, parseTransaction),
```

Comment:
The advantage of Lodash is that it can optimize for each browser/interpreter. Using `isEmpty` and `isString` is a good idea.